### PR TITLE
[SWR] Refactoring after `gophertelekomcloud` update 

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
## PR Checklist

* [x] Tests added/passed.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestSwrRepositoryV2_basic
--- PASS: TestSwrRepositoryV2_basic (55.84s)
PASS


Process finished with the exit code 0

=== RUN   TestSwrOrganizationV2_basic
--- PASS: TestSwrOrganizationV2_basic (79.30s)
=== RUN   TestSwrOrganizationV2_validateName
--- PASS: TestSwrOrganizationV2_validateName (18.44s)
PASS

Process finished with the exit code 0

=== RUN   TestSwrOrganizationPermissionsV2_basic
--- PASS: TestSwrOrganizationPermissionsV2_basic (63.26s)
PASS


Process finished with the exit code 0

=== RUN   TestSwrDomainV2Basic
--- PASS: TestSwrDomainV2Basic (67.57s)
PASS

Process finished with the exit code 0


=== RUN   TestSwrRepositoryV2_import
--- PASS: TestSwrRepositoryV2_import (52.99s)
PASS


Process finished with the exit code 0

=== RUN   TestSwrOrganizationV2_import
--- PASS: TestSwrOrganizationV2_import (55.60s)
PASS


Process finished with the exit code 0
```

### CODE REVIEW

**Review by Aloento — APPROVED** [review]

File: opentelekomcloud/services/swr/resource_opentelekomcloud_swr_domain_v2.go [hunk 1]
```diff
@@ -114,15 +112,20 @@ func resourceSwrDomainRead(_ context.Context, d *schema.ResourceData, meta inter
 		return fmterr.Errorf(ClientError, err)
 	}
 
-	domain, err := domains.Get(client, organization(d), repository(d), d.Id()).Extract()
+	opts := domains.GetOpts{
+		Namespace:    d.Get("organization").(string),
+		Repository:   d.Get("repository").(string),
+		AccessDomain: d.Get("access_domain").(string),
```
Comment (artem-lifshits) [inline opentelekomcloud/services/swr/resource_opentelekomcloud_swr_domain_v2.go:118]: `AccessDomain` wasn't present in previous version, did something change?

File: opentelekomcloud/services/swr/resource_opentelekomcloud_swr_domain_v2.go [hunk 2]
```diff
@@ -166,7 +171,12 @@ func resourceSwrDomainDelete(_ context.Context, d *schema.ResourceData, meta int
 		return fmterr.Errorf(ClientError, err)
 	}
 
-	err = domains.Delete(client, organization(d), repository(d), d.Id()).ExtractErr()
+	opts := domains.GetOpts{
+		Namespace:    d.Get("organization").(string),
+		Repository:   d.Get("repository").(string),
+		AccessDomain: d.Get("access_domain").(string),
+	}
+	err = domains.Delete(client, opts)
```
Comment (artem-lifshits) [inline opentelekomcloud/services/swr/resource_opentelekomcloud_swr_domain_v2.go:179]: Same here

File: opentelekomcloud/services/swr/resource_opentelekomcloud_swr_repository_v2.go [hunk 3]
```diff
@@ -153,11 +154,13 @@ func resourceRepositoryUpdate(ctx context.Context, d *schema.ResourceData, meta
 	}
 
 	opts := repositories.UpdateOpts{
+		Namespace:   d.Get("organization").(string),
+		Repository:  d.Get("repository").(string),
```
Comment (artem-lifshits) [inline opentelekomcloud/services/swr/resource_opentelekomcloud_swr_repository_v2.go:158]: `Repository` wasn't present before

**Review by vladimirvshivkov — APPROVED** [review]

File: opentelekomcloud/services/swr/resource_opentelekomcloud_swr_domain_v2.go (hunk 1 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/services/swr/resource_opentelekomcloud_swr_domain_v2.go:118]: it passed directly to function now it in opts

File: opentelekomcloud/services/swr/resource_opentelekomcloud_swr_domain_v2.go (hunk 2 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/services/swr/resource_opentelekomcloud_swr_domain_v2.go:179]: same

File: opentelekomcloud/services/swr/resource_opentelekomcloud_swr_repository_v2.go (hunk 3 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/services/swr/resource_opentelekomcloud_swr_repository_v2.go:158]: before:
repositories.Get(client, organization(d), d.Id()).Extract()
 organization(d) = d.Get("organization").(string)
 d.Id() = d.Get("repository").(string)
 
 after 
repositories.Update(client, repositories.UpdateOpts{
		Namespace:   d.Get("organization").(string),
		Repository:  d.Get("repository").(string),})

**Review by artem-lifshits — APPROVED** [review]

**Review by bakhterets — COMMENTED** [review]
viewed

**Pull request merged**